data/shared/src/main/scala/sigma/serialization/MethodCallSerializer.scala

```diff
@@ -60,6 +59,16 @@ case class MethodCallSerializer(cons: (Value[SType], SMethod, IndexedSeq[Value[S
       }
 
     val specMethod = method.specializeFor(obj.tpe, types)
+
+    var isUsingBlockchainContext = specMethod.objType == SContextMethods && (
+      method.name == SContextMethods.headersMethod.name ||
+        method.name == SContextMethods.preHeaderMethod.name ||
+        method.name == SContextMethods.heightMethod.name ||
+        method.name == SContextMethods.lastBlockUtxoRootHashMethod.name ||
+        method.name == SContextMethods.minerPubKeyMethod.name
+      )
```

Comment:
I think we can slice a bit of performance out of this code. 
The lines like `SContextMethods.headersMethod.name` generate at least 2 method calls. 
I suggest to cache all of them in Array[String] and search in there using tight loop.

Reply:
Fixed in https://github.com/ScorexFoundation/sigmastate-interpreter/pull/929/commits/22f058625a15767926d53b23ce5812a0e78a7845

Please take a look.